test/test_utils.jl

```diff
@@ -349,6 +349,127 @@ function ad_constant_allocs_heuristic(
     pb2(Δ2_val)
     allocs_2 = count_allocs(pb2, Δ2 === nothing ? out2 : Δ2)
 
-    pullback_heuristic = allocs_1 == allocs_2
-    return primal_heuristic, forwards_heuristic, pullback_heuristic
+    return primal_heuristic, forwards_heuristic, (allocs_1, allocs_2)
+end
+
+"""
+    test_zygote_perf_heuristic(
+        f, name::String, args1, args2, passes, Δ1=nothing, Δ2=nothing
+    )
+
+Executes `ad_constant_allocs_heuristc(f, args1, args2; Δ1, Δ2)` and creates a testset out
+of the results.
+`passes` is a 3-tuple of booleans. `passes[1]` indicates whether the checks should pass
+for `f(args1...)` etc, `passes[2]` for `Zygote.pullback(f, args1...)`, and `passes[3]` the
+pullback generated by `Zygote.pullback(f, args1...)`. This is useful when it is known that
+some of the tests fail and a fix isn't immediately available.
+"""
+function test_zygote_perf_heuristic(
+    f, name::String, args1, args2, passes, Δ1=nothing, Δ2=nothing
+)
+    @testset "$name" begin
+        primal, fwd, pb = ad_constant_allocs_heuristic(f, args1, args2; Δ1, Δ2)
+        if passes[1]
+            @test primal[1] == primal[2]
+        else
+            @test_broken primal[1] == primal[2]
+        end
```

Comment:
I guess you can simplify those as
```suggestion
             @test primal[1] == primal[2] broken=!passes[1]
```

Reply:
Arf, I thought it was older than this